src/main/java/htsjdk/tribble/gff/Gff3Codec.java

```diff
@@ -198,6 +180,31 @@ public Gff3Feature decode(final LineIterator lineIterator) throws IOException {
         return attributes;
     }
 
+    static private Gff3BaseData parseLine(final String line) {
+        final List<String> splitLine = ParsingUtils.split(line, FIELD_DELIMITER);
+
+        if (splitLine.size() != NUM_FIELDS) {
+            throw new TribbleException("Found an invalid number of columns in the given Gff3 file- Given: " + splitLine.size() + " Expected: " + NUM_FIELDS + " : " + line);
+        }
+
+        try {
+            final String contig = URLDecoder.decode(splitLine.get(CHROMOSOME_NAME_INDEX), "UTF-8");
+            final String source = URLDecoder.decode(splitLine.get(ANNOTATION_SOURCE_INDEX), "UTF-8");
+            final String type = URLDecoder.decode(splitLine.get(FEATURE_TYPE_INDEX), "UTF-8");
+            final int start = Integer.parseInt(splitLine.get(START_LOCATION_INDEX));
+            final int end = Integer.parseInt(splitLine.get(END_LOCATION_INDEX));
+            final Double score = splitLine.get(SCORE_INDEX).equals(".") ? -1 : Double.parseDouble(splitLine.get(SCORE_INDEX));
+            final int phase = splitLine.get(GENOMIC_PHASE_INDEX).equals(".") ? -1 : Integer.parseInt(splitLine.get(GENOMIC_PHASE_INDEX));
+            final Strand strand = Strand.decode(splitLine.get(GENOMIC_STRAND_INDEX));
+            final Map<String, String> attributes = parseAttributes(splitLine.get(EXTRA_FIELDS_INDEX));
+            return new Gff3BaseData(contig, source, type, start, end, score, strand, phase, attributes);
+        } catch (final NumberFormatException ex ) {
+            throw new TribbleException("Cannot read integer value for start/end position!", ex);
```

Comment:
You should include the at least the entirety of the line here, and I would probably try to include the line number as well.

Reply:
done